src/lib/storage/table.cpp

```diff
@@ -86,11 +88,43 @@ const std::string& Table::column_type(ColumnID column_id) const {
 
 Chunk& Table::get_chunk(ChunkID chunk_id) {
   DebugAssert(chunk_id < _chunks.size(), "No chunk with given ID");
+  std::shared_lock read_lock(_chunk_access);
+  return *_chunks[chunk_id];
+}
+
+const Chunk& Table::get_chunk(ChunkID chunk_id) const {
+  DebugAssert(chunk_id < _chunks.size(), "No chunk with given ID");
+  std::shared_lock read_lock(_chunk_access);
   return *_chunks[chunk_id];
 }
 
-const Chunk& Table::get_chunk(ChunkID chunk_id) const { return get_chunk(chunk_id); }
+void Table::compress_chunk(ChunkID chunk_id) {
+  auto& chunk = get_chunk(chunk_id);
+
+  // create structures and lambda function
+  std::vector<std::thread> threads;
+  std::vector<std::shared_ptr<BaseSegment>> compressed_segments(chunk.column_count());
+  auto compress = [&compressed_segments](auto type, auto segment, auto segment_id) {
+    compressed_segments.at(segment_id) = make_shared_by_data_type<BaseSegment, DictionarySegment>(type, segment);
+  };
+
+  // start thread for each segment
+  for (ColumnID column_id(0); column_id < chunk.size(); ++column_id) {
```

Comment:
Again comparing column ids with the length of a chunk. The only reason this does not break is because by chance, the test has as many columns as rows in a chunk.

Reply:
So using a size_t and creating a new ColumnID when indexing is desired here?

Reply:
Suggestion: `for (auto column_id = ColumnID{0}; column_id < chunk.column_count(); ++column_id`